From 0b21384bcd5ea71885dde8ebcb4c4d1bfd6f0187 Mon Sep 17 00:00:00 2001
From: Paul Cercueil <paul.cercueil@analog.com>
Date: Mon, 3 Jul 2017 13:07:02 +0200
Subject: [PATCH] Fix public functions visibility on Windows

Signed-off-by: Paul Cercueil <paul.cercueil@analog.com>
---
 libsigrokdecode.h | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/libsigrokdecode.h b/libsigrokdecode.h
index 03a5de0..ed25e89 100644
--- a/libsigrokdecode.h
+++ b/libsigrokdecode.h
@@ -108,8 +108,10 @@ enum srd_loglevel {
 /* Marks public libsigrokdecode API symbols. */
 #ifndef _WIN32
 #define SRD_API __attribute__((visibility("default")))
+#elif defined(LIBSIGROKDECODE_EXPORT)
+#define SRD_API __declspec(dllexport)
 #else
-#define SRD_API
+#define SRD_API __declspec(dllimport)
 #endif
 
 /* Marks private, non-public libsigrokdecode symbols (not part of the API). */
-- 
2.13.2

